pyzx/extract.py

```diff
@@ -222,6 +229,115 @@ def greedy_reduction(m: Mat2) -> Optional[List[Tuple[int,int]]]:
         indices.remove(control)
     return result
 
+
+def flat_indices(m: Mat2, indices: list[int]) -> Tuple[List[Tuple[int, int]], int]:
+    if len(indices) == 0:
+        return [], -1
+    rows = {i: m.data[i] for i in indices}
+    weights = {i: sum(r) for i, r in rows.items()}
+    result = []
+    next_indices = []
+    while len(indices) > 1:
+        best = (-1, -1)
+        reduction = -10000
+        for i in indices:
+            for j in indices:
+                if j <= i: continue
+                w = sum(xor_rows(rows[i], rows[j]))
+                if weights[i] - w > reduction:
+                    best = (j, i)  # "Add row j to i"
+                    reduction = weights[i] - w
+                if weights[j] - w > reduction:
+                    best = (i, j)
+                    reduction = weights[j] - w
+        result.append(best)
+        control, target = best
+        rows[target] = xor_rows(rows[control], rows[target])
+        weights[target] = weights[target] - reduction
+        indices.remove(control)
+        indices.remove(target)
+        next_indices.append(target)
+        if len(indices) <= 1:
+            if len(indices) == 1:
+                next_indices.append(indices[0])
+            indices = next_indices
+            next_indices = []
+    return result, indices[0]
+
+
+def greedy_reduction2(m: Mat2) -> Optional[List[Tuple[int, int]]]:
```

Comment:
Perhaps have a more descriptive name than just adding a 2 at the end? Or at least make more clear in the docstring how it differs.